v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/columnboundary/ColumnForBoundaryQuery.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.columnboundary;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.range.PartitionColumn;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.range.Range;
+import java.io.Serializable;
+import javax.annotation.Nullable;
+
+/**
+ * Intermediate data structure generated during PTransforms to indicate a column for which a
+ * boundary (min, max) is requested.
+ */
+@AutoValue
+public abstract class ColumnForBoundaryQuery implements Serializable {
+
+  /**
+   * @return column details.
+   */
+  public abstract PartitionColumn partitionColumn();
+
+  /**
+   * @return Parent Range. Null indicates first column for splitting. Defaults to Null.
+   */
+  @Nullable
+  public abstract Range parentRange();
```

Comment:
can you elaborate on the docs here. It is not clear from the method name.

Reply:
Added more details. Thanks.